src/conn/conn_api.c

```diff
@@ -1285,6 +1285,35 @@ __conn_rollback_to_stable(WT_CONNECTION *wt_conn, const char *config)
 err:	API_END_RET(session, ret);
 }
 
+/*
+ * __conn_calibrate_ticks --
+ *	Calibrate a ratio from rdtsc ticks to nanoseconds.
+ */
+static void
+__conn_calibrate_ticks(WT_SESSION_IMPL *session)
+{
+#if (defined __i386) || (defined __amd64)
+	struct timespec start, stop;
+	uint64_t diff_nsec, diff_tsc, tsc_start, tsc_stop;
+	volatile uint64_t i;
+
+	__wt_epoch(session, &start);
+	tsc_start = __wt_rdtsc(session);
+	/*
+	 * This needs to be CPU intensive and large enough.
+	 */
+	for (i = 0; i < WT_MILLION; i++)
+		;
+	tsc_stop = __wt_rdtsc(session);
+	__wt_epoch(session, &stop);
+	diff_nsec = WT_TIMEDIFF_NS(stop, start);
+	diff_tsc = tsc_stop - tsc_start;
+	S2C(session)->tsc_nsec_ratio = (double)diff_tsc/(double)diff_nsec;
+#else
+	S2C(session)->tsc_nsec_ratio = 1.0;
+#endif
+}
+
```

Comment:
@sueloverso, I like the idea of using a loop for determining the ratio of clock ticks to nanoseconds. Based on my investigation, there is no cross-platform way to do this conversion reliably: for example the method suggested for OS X is broken and even the stack overflow community had no answer.

I do not know if one million iterations is sufficient though.

Reply:
I ran with 1M, 10M and 1B and they all computed the same ratio out to the 4th decimal place.  There was a noticeable delay in `wiredtiger_open` with 1B.